# Project 1.
Olimpic games dataset analysis.

## 1. Read data
Function `read_data` meges datasets from files of specialised type. In our case datasets are in files `.csv`.

In [74]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np
from scipy import stats


In [2]:
import os
def read_data(file_type, folder):
  files_names = []
  for file in os.listdir(folder):
    if file.endswith(file_type):
      files_names.append(os.path.join(folder, file))
  data_table = pd.concat(map(lambda x: pd.read_table(x, sep=',') , files_names), ignore_index=True)
  return data_table
        

In [3]:
file_type = '.csv'
folder = '/content/drive/MyDrive/Stat_data'
data = read_data(file_type, folder)

In [52]:
data

ID                          Name Sex   Age  Height  Weight  \
0        79588         Rodolfo Mijares Garza   M  22.0   180.0    75.0   
1        79589              Zlatica Mijatovi   F  26.0     NaN     NaN   
2        79590                  Pieter Mijer   M  47.0     NaN     NaN   
3        79591                     Vasa Miji   M  27.0   186.0    80.0   
4        79591                     Vasa Miji   M  31.0   186.0    80.0   
...        ...                           ...  ..   ...     ...     ...   
271110  102340           Sigrid Romero Duque   F  19.0   164.0    57.0   
271111  102340           Sigrid Romero Duque   F  19.0   164.0    57.0   
271112  102341        Susana Romero Steensma   F  17.0   173.0    64.0   
271113  102342  Yasmani Jess Romero Rodrguez   M  23.0     NaN     NaN   
271114  102343          Yoel Romero Palacios   M  23.0   175.0    84.0   

                         Team  NOC        Games    Year  Season       City  \
0                      Mexico  MEX  1960 Summer  1960.0  Summer       Roma   
1                  Yugoslavia  YUG  1948 Summer  1948.0  Summer     London   
2                 Netherlands  NED  1928 Summer  1928.0  Summer  Amsterdam   
3       Serbia and Montenegro  SCG  2000 Summer  2000.0  Summer     Sydney   
4       Serbia and Montenegro  SCG  2004 Summer  2004.0  Summer     Athina   
...                       ...  ...          ...     ...     ...        ...   
271110               Colombia  COL  2008 Summer  2008.0  Summer    Beijing   
271111               Colombia  COL  2008 Summer  2008.0  Summer    Beijing   
271112                  Spain  ESP  2008 Summer  2008.0  Summer    Beijing   
271113                   Cuba  CUB  2012 Summer  2012.0  Summer     London   
271114                   Cuba  CUB  2000 Summer  2000.0  Summer     Sydney   

                Sport                                         Event   Medal  
0           Athletics                     Athletics Men's Decathlon     NaN  
1          Gymnastics            Gymnastics Women's Team All-Around     NaN  
2             Fencing                Fencing Men's epee, Individual     NaN  
3          Volleyball                   Volleyball Men's Volleyball    Gold  
4          Volleyball                   Volleyball Men's Volleyball     NaN  
...               ...                                           ...     ...  
271110        Archery                    Archery Women's Individual     NaN  
271111        Archery                          Archery Women's Team     NaN  
271112        Sailing             Sailing Women's One Person Dinghy     NaN  
271113  Weightlifting              Weightlifting Men's Bantamweight     NaN  
271114      Wrestling  Wrestling Men's Light-Heavyweight, Freestyle  Silver  

[271115 rows x 15 columns]

##2. Problems with data.


**Sex**

There are not only 'F' and 'M' in our data (there is 'G')))) 

I think we should change it into 'M' (I really hope that Pavel's and Anantoly's sex is 'mail').

The are also two empty strings with 'NaN' everywhere. 

In [53]:
data.Sex.unique()
data[data.Sex.isna()]

ID         Name  Sex  Age  Height  Weight Team  NOC Games  Year  \
90194   124516          NaN  NaN  NaN     NaN     NaN  NaN  NaN   NaN   NaN   
112604  113716  Pietro Spec  NaN  NaN     NaN     NaN  NaN  NaN   NaN   NaN   

       Season City Sport Event Medal  
90194     NaN  NaN   NaN   NaN   NaN  
112604    NaN  NaN   NaN   NaN   NaN

In [54]:
data[data.Sex == 'G']

ID               Name Sex   Age  Height  Weight            Team  NOC  \
42  79609         Pavel Mike   G  22.0   182.0    79.0  Czechoslovakia  TCH   
74  79630  Anatoly Mikhaylin   G  37.0     NaN     NaN          Russia  RUS   

          Games    Year  Season     City     Sport  \
42  1972 Summer  1972.0  Summer   Munich  Handball   
74  1996 Summer  1996.0  Summer  Atlanta   Sailing   

                                Event   Medal  
42            Handball Men's Handball  Silver  
74  Sailing Mixed Two Person Keelboat     NaN

In [55]:
corrected_data = data.drop(labels=[90194, 112604], axis=0)
corrected_data.at[[42, 74], 'Sex'] = 'M'
corrected_data.Sex.unique()

array(['M', 'F'], dtype=object)

**Age**

There are some strange ages in table.

In [56]:
Ages  = corrected_data.Age.unique()
Ages

array([ 22.,  26.,  47.,  27.,  31.,  25.,  30.,  34.,  38.,  28.,  nan,
        17.,  21.,  32.,  24.,  29.,  33.,  20.,  19.,  23.,  37.,  35.,
        60.,  16.,  36.,  39.,  46.,  15.,  18.,  48.,  40.,  44.,  41.,
        45.,  49.,  53.,  57.,  61.,  65.,  56.,  52.,  42.,  14.,  43.,
        51.,  13.,  50.,  54.,  67.,  55.,  58.,  63.,  69.,  12.,  66.,
        76.,  77.,  59.,  62., 240.,  64.,  71.,  74.,  70.,  68.,  72.,
        88.,  11.,  97.,  81.,  73.,  80.,  10.,  96.,  75.,  84.])

In [57]:
wrong_age = corrected_data.query("Age == 240")
wrong_age

ID                        Name Sex    Age  Height  Weight     Team  \
22820  23459  Flicien Jules mile Courbet   M  240.0     NaN     NaN  Belgium   

       NOC        Games    Year  Season       City     Sport  \
22820  BEL  1912 Summer  1912.0  Summer  Stockholm  Swimming   

                                        Event Medal  
22820  Swimming Men's 200 metres Breaststroke   NaN

Something strange happened with Flicien Jules mile Courbet in one string. We will change his age from 240 to 24.

In [58]:
corrected_data.loc[corrected_data['Age']==240, ['Age']] = 24.0
corrected_data.loc[corrected_data['Name']=='Flicien Jules mile Courbet']

ID                        Name Sex   Age  Height  Weight     Team  \
22818  23459  Flicien Jules mile Courbet   M  20.0     NaN     NaN  Belgium   
22819  23459  Flicien Jules mile Courbet   M  24.0     NaN     NaN  Belgium   
22820  23459  Flicien Jules mile Courbet   M  24.0     NaN     NaN  Belgium   
22821  23459  Flicien Jules mile Courbet   M  24.0     NaN     NaN  Belgium   
22822  23459  Flicien Jules mile Courbet   M  32.0     NaN     NaN  Belgium   
22823  23459  Flicien Jules mile Courbet   M  32.0     NaN     NaN  Belgium   

       NOC        Games    Year  Season       City       Sport  \
22818  BEL  1908 Summer  1908.0  Summer     London    Swimming   
22819  BEL  1912 Summer  1912.0  Summer  Stockholm  Water Polo   
22820  BEL  1912 Summer  1912.0  Summer  Stockholm    Swimming   
22821  BEL  1912 Summer  1912.0  Summer  Stockholm    Swimming   
22822  BEL  1920 Summer  1920.0  Summer  Antwerpen    Swimming   
22823  BEL  1920 Summer  1920.0  Summer  Antwerpen    Swimming   

                                        Event   Medal  
22818  Swimming Men's 200 metres Breaststroke     NaN  
22819             Water Polo Men's Water Polo  Bronze  
22820  Swimming Men's 200 metres Breaststroke     NaN  
22821  Swimming Men's 400 metres Breaststroke     NaN  
22822  Swimming Men's 200 metres Breaststroke     NaN  
22823  Swimming Men's 400 metres Breaststroke     NaN

**Height**

In [59]:
Heights = corrected_data.Height.unique()

There is one strange height 340.

In [60]:
wrong_height = corrected_data.query("Height == 340")
wrong_height

ID                             Name Sex   Age  Height  Weight  \
23009  23549  Kirsty Leigh Coventry (-Seward)   F  28.0   340.0    64.0   

           Team  NOC        Games    Year  Season    City     Sport  \
23009  Zimbabwe  ZIM  2012 Summer  2012.0  Summer  London  Swimming   

                                               Event Medal  
23009  Swimming Women's 200 metres Individual Medley   NaN

In [61]:
corrected_data.loc[corrected_data['Height']==340, ['Height']] = 176.0 #watched from other strings in table
corrected_data.loc[corrected_data['Name']=='Kirsty Leigh Coventry (-Seward)']

ID                             Name Sex   Age  Height  Weight  \
22996  23549  Kirsty Leigh Coventry (-Seward)   F  16.0   176.0    64.0   
22997  23549  Kirsty Leigh Coventry (-Seward)   F  16.0   176.0    64.0   
22998  23549  Kirsty Leigh Coventry (-Seward)   F  16.0   176.0    64.0   
22999  23549  Kirsty Leigh Coventry (-Seward)   F  16.0   176.0    64.0   
23000  23549  Kirsty Leigh Coventry (-Seward)   F  20.0   176.0    64.0   
23001  23549  Kirsty Leigh Coventry (-Seward)   F  20.0   176.0    64.0   
23002  23549  Kirsty Leigh Coventry (-Seward)   F  20.0   176.0    64.0   
23003  23549  Kirsty Leigh Coventry (-Seward)   F  24.0   176.0    64.0   
23004  23549  Kirsty Leigh Coventry (-Seward)   F  24.0   176.0    64.0   
23005  23549  Kirsty Leigh Coventry (-Seward)   F  24.0   176.0    64.0   
23006  23549  Kirsty Leigh Coventry (-Seward)   F  24.0   176.0    64.0   
23007  23549  Kirsty Leigh Coventry (-Seward)   F  28.0   176.0    64.0   
23008  23549  Kirsty Leigh Coventry (-Seward)   F  28.0   176.0    64.0   
23009  23549  Kirsty Leigh Coventry (-Seward)   F  28.0   176.0    64.0   
23010  23549  Kirsty Leigh Coventry (-Seward)   F  32.0   176.0    64.0   
23011  23549  Kirsty Leigh Coventry (-Seward)   F  32.0   176.0    64.0   

           Team  NOC        Games    Year  Season            City     Sport  \
22996  Zimbabwe  ZIM  2000 Summer  2000.0  Summer          Sydney  Swimming   
22997  Zimbabwe  ZIM  2000 Summer  2000.0  Summer          Sydney  Swimming   
22998  Zimbabwe  ZIM  2000 Summer  2000.0  Summer          Sydney  Swimming   
22999  Zimbabwe  ZIM  2000 Summer  2000.0  Summer          Sydney  Swimming   
23000  Zimbabwe  ZIM  2004 Summer  2004.0  Summer          Athina  Swimming   
23001  Zimbabwe  ZIM  2004 Summer  2004.0  Summer          Athina  Swimming   
23002  Zimbabwe  ZIM  2004 Summer  2004.0  Summer          Athina  Swimming   
23003  Zimbabwe  ZIM  2008 Summer  2008.0  Summer         Beijing  Swimming   
23004  Zimbabwe  ZIM  2008 Summer  2008.0  Summer         Beijing  Swimming   
23005  Zimbabwe  ZIM  2008 Summer  2008.0  Summer         Beijing  Swimming   
23006  Zimbabwe  ZIM  2008 Summer  2008.0  Summer         Beijing  Swimming   
23007  Zimbabwe  ZIM  2012 Summer  2012.0  Summer          London  Swimming   
23008  Zimbabwe  ZIM  2012 Summer  2012.0  Summer          London  Swimming   
23009  Zimbabwe  ZIM  2012 Summer  2012.0  Summer          London  Swimming   
23010  Zimbabwe  ZIM  2016 Summer  2016.0  Summer  Rio de Janeiro  Swimming   
23011  Zimbabwe  ZIM  2016 Summer  2016.0  Summer  Rio de Janeiro  Swimming   

                                               Event   Medal  
22996           Swimming Women's 50 metres Freestyle     NaN  
22997          Swimming Women's 100 metres Freestyle     NaN  
22998         Swimming Women's 100 metres Backstroke     NaN  
22999  Swimming Women's 200 metres Individual Medley     NaN  
23000         Swimming Women's 100 metres Backstroke  Silver  
23001         Swimming Women's 200 metres Backstroke    Gold  
23002  Swimming Women's 200 metres Individual Medley  Bronze  
23003         Swimming Women's 100 metres Backstroke  Silver  
23004         Swimming Women's 200 metres Backstroke    Gold  
23005  Swimming Women's 200 metres Individual Medley  Silver  
23006  Swimming Women's 400 metres Individual Medley  Silver  
23007         Swimming Women's 100 metres Backstroke     NaN  
23008         Swimming Women's 200 metres Backstroke     NaN  
23009  Swimming Women's 200 metres Individual Medley     NaN  
23010         Swimming Women's 100 metres Backstroke     NaN  
23011         Swimming Women's 200 metres Backstroke     NaN

**Weight**

In [ ]:
Weights = corrected_data.Weight.unique()
Weights

In [63]:
corrected_data.loc[corrected_data['Weight']>180]

ID                           Name Sex   Age  Height  Weight  \
78509   118869  Christopher J. "Chris" Taylor   M  22.0   196.0   182.0   
78510   118869  Christopher J. "Chris" Taylor   M  22.0   196.0   182.0   
91811   103159              Aytami Ruano Vega   M  27.0   200.0   198.0   
187565   38075                  Marek Galiski   M  29.0   200.0   190.0   
203494   12177              Ricardo Blas, Jr.   M  21.0   183.0   214.0   
203495   12177              Ricardo Blas, Jr.   M  25.0   183.0   214.0   

                 Team  NOC        Games    Year  Season     City      Sport  \
78509   United States  USA  1972 Summer  1972.0  Summer   Munich  Wrestling   
78510   United States  USA  1972 Summer  1972.0  Summer   Munich  Wrestling   
91811           Spain  ESP  2004 Summer  2004.0  Summer   Athina       Judo   
187565         Poland  POL  1980 Summer  1980.0  Summer   Moskva  Wrestling   
203494           Guam  GUM  2008 Summer  2008.0  Summer  Beijing       Judo   
203495           Guam  GUM  2012 Summer  2012.0  Summer   London       Judo   

                                                 Event   Medal  
78509   Wrestling Men's Super-Heavyweight, Greco-Roman     NaN  
78510     Wrestling Men's Super-Heavyweight, Freestyle  Bronze  
91811                           Judo Men's Heavyweight     NaN  
187565  Wrestling Men's Super-Heavyweight, Greco-Roman     NaN  
203494                          Judo Men's Heavyweight     NaN  
203495                          Judo Men's Heavyweight     NaN

I googled this 214 kg guy, he really exists...
So I didn't correct anything there.

**Team**

In [16]:
Teams = corrected_data.Team.unique()
Teams

array(['Mexico', 'Yugoslavia', 'Netherlands', ..., 'Cicely-1', "Ma'Lindo",
       'Orion'], dtype=object)

We will search for mistakes in 'Team' data using regular expressions. Firstly, let's look at teams with simple mistakes (mistakes which can be written by RegEx). Such mistakes are numbers (arabic or latin) in team name.

In [17]:
mistakes = data[data['Team'].str.contains('[0-9]|III|II|VI|IV|IX|XI| V$| X$')==True]
mistakes 

ID                            Name Sex   Age  Height  Weight  \
70       79627                 Isidoros Mikhas   M   NaN     NaN     NaN   
114      79655  Stanislav Sergeyevich Mikheyev   M  20.0   172.0    66.0   
120      79659        Dimitrios Mikhitsopoulos   M   NaN     NaN     NaN   
267      79736                 Ioannis Milakas   M   NaN     NaN     NaN   
334      79767             Hctor Milberg Bossi   M  23.0     NaN     NaN   
...        ...                             ...  ..   ...     ...     ...   
270896  102234                Samuele Romanini   M  29.0   187.0   105.0   
270897  102234                Samuele Romanini   M  33.0   187.0   105.0   
270902  102236             Constantin Romaniuc   M  22.0     NaN     NaN   
270941  102259                  Iryna Romanova   F  21.0   168.0    54.0   
270942  102259                  Iryna Romanova   F  25.0   168.0    54.0   

                                                   Team  NOC        Games  \
70      Large boat, Central Naval Prep School "Poros"-1  GRE  1906 Summer   
114                                            Russia-2  RUS  2010 Winter   
120                                       Thessalonki-1  GRE  1906 Summer   
267                  Phalainis ton Thorichtou "Hydra"-2  GRE  1906 Summer   
334                                         Argentina-2  ARG  1928 Winter   
...                                                 ...  ...          ...   
270896                                          Italy-1  ITA  2006 Winter   
270897                                          Italy-1  ITA  2010 Winter   
270902                                        Romania-2  ROU  1976 Winter   
270941                                        Ukraine-1  UKR  1994 Winter   
270942                                        Ukraine-1  UKR  1998 Winter   

          Year  Season          City           Sport  \
70      1906.0  Summer        Athina          Rowing   
114     2010.0  Winter     Vancouver            Luge   
120     1906.0  Summer        Athina        Football   
267     1906.0  Summer        Athina          Rowing   
334     1928.0  Winter  Sankt Moritz       Bobsleigh   
...        ...     ...           ...             ...   
270896  2006.0  Winter        Torino       Bobsleigh   
270897  2010.0  Winter     Vancouver       Bobsleigh   
270902  1976.0  Winter     Innsbruck       Bobsleigh   
270941  1994.0  Winter   Lillehammer  Figure Skating   
270942  1998.0  Winter        Nagano  Figure Skating   

                                         Event   Medal  
70      Rowing Men's 17-Man Naval Rowing Boats    Gold  
114                 Luge Mixed (Men)'s Doubles     NaN  
120                    Football Men's Football  Bronze  
267     Rowing Men's 17-Man Naval Rowing Boats  Silver  
334                  Bobsleigh Men's Four/Five     NaN  
...                                        ...     ...  
270896                    Bobsleigh Men's Four     NaN  
270897                     Bobsleigh Men's Two     NaN  
270902                    Bobsleigh Men's Four     NaN  
270941        Figure Skating Mixed Ice Dancing     NaN  
270942        Figure Skating Mixed Ice Dancing     NaN  

[6550 rows x 15 columns]

We see that there are a lot of strings with incorrect team name. We could remove these strings (already 6.5k strings on stage with correcting via RegEx) or we can create dictionary for translation NOC -> team. I'm lazy for this)

I will take countries from column 'NOC' if I need countries in following tasks (for axample, task 11).

#3. Youngest athlets.

In [18]:
min(corrected_data[corrected_data['Sex']=='M'].Age)

10.0

In [19]:
min(corrected_data[corrected_data['Sex']=='F'].Age)

11.0

**Answer**

The youngest boy's age is 10. The youngest girl's age is 11.

##4. Mean of 'Height' for man and woman.

In [20]:
np.mean(corrected_data[corrected_data['Sex']=='M'].Height), np.std(corrected_data[corrected_data['Sex']=='M'].Height) 

(178.85861358000614, 9.360312373726755)

In [21]:
np.mean(corrected_data[corrected_data['Sex']=='F'].Height), np.std(corrected_data[corrected_data['Sex']=='F'].Height)

(167.8397275115392, 8.778398158593662)

##5. Mean of 'Height' for female tennis players on the Olympic games 2000.

In [28]:
female_tennis = corrected_data.query("Sport=='Tennis'  and Sex=='F' and Year==2000")
np.mean(female_tennis.Height), np.std(female_tennis.Height)

(171.792, 6.4327860216239126)

**Answer**

mean hight = 171.8

standart deviation = 6.4

##6. The haviest sportsman 2006##

In [29]:
olympic_2006 = corrected_data.query("Year==2006")
olympic_2006[olympic_2006['Weight']==max(olympic_2006.Weight)]

ID               Name Sex   Age  Height  Weight     Team  NOC  \
233864  4476  Patrick R. Antaki   M  41.0   185.0   127.0  Lebanon  LIB   

              Games    Year  Season    City     Sport  \
233864  2006 Winter  2006.0  Winter  Torino  Skeleton   

                          Event Medal  
233864  Skeleton Men's Skeleton   NaN

The name of the haviest sportsman is Patrick R. Antaki. And his kind of sport is **Skeleton**!

##7. Women with gold medals 1980-2010##

In [32]:
Gold_women = corrected_data[corrected_data['Sex']=='F'].query("Year>=1980  and Year<=2010 and Medal=='Gold'")
Gold_women.Medal.count()

2249

**Answer**

Women got **2249** gold medals!

##8. John Aalberg##

In [40]:
corrected_data[corrected_data['Name']=='John Aalberg'].Year.value_counts()

1992.0    4
1994.0    4
Name: Year, dtype: int64

**Answer**
He took part in Olympic games in 1992 and 1994.
**4 times in every year!**

##9. Age groups.##

In [51]:
olympic_2008 = corrected_data[corrected_data['Year']==2008]
Age_groups = {'[15;25)': len(olympic_2008.query("Age>=15  and Age<25").ID.unique()),
              '[25;35)': len(olympic_2008.query("Age>=25  and Age<35").ID.unique()),
              '[35;45)':len(olympic_2008.query("Age>=35  and Age<45").ID.unique()),
              '[45;55]':len(olympic_2008.query("Age>=45  and Age<=55").ID.unique())}
Age_groups

{'[15;25)': 4786, '[25;35)': 5382, '[35;45)': 630, '[45;55]': 78}

**Answer**

The most represented group is [25;35).

The less represented group is [45;55).

##10. Kind of sports 2002 and 1994.##

In [65]:
kind_2002 = corrected_data.query("Year==2002").Sport.unique()
kind_1994 = corrected_data.query("Year==1994").Sport.unique()
len(kind_2002)-len(kind_1994)

3

**Answer**

There are **3** more sports in 2002.

##11. Top 3##

Here, as I mentioned above, I will use NOCs instead Countries.

In [72]:
gold_winter = corrected_data.query("Medal=='Gold' and Season=='Winter'").value_counts('NOC').to_frame().head(3)
gold_summer = corrected_data.query("Medal=='Gold' and Season=='Summer'").value_counts('NOC').to_frame().head(3)
silver_winter = corrected_data.query("Medal=='Silver' and Season=='Winter'").value_counts('NOC').to_frame().head(3)
silver_summer = corrected_data.query("Medal=='Silver' and Season=='Summer'").value_counts('NOC').to_frame().head(3)
bronze_winter = corrected_data.query("Medal=='Bronze' and Season=='Winter'").value_counts('NOC').to_frame().head(3)
bronze_summer = corrected_data.query("Medal=='Bronze' and Season=='Summer'").value_counts('NOC').to_frame().head(3)
reiting = pd.concat([gold_winter, silver_winter, bronze_winter, gold_summer, silver_summer, bronze_summer], 
                    sort=False, keys=['gold_winter', 'silver_winter', 'bronze_winter', 'gold_summer', 'silver_summer', 'bronze_summer'])
reiting

0
              NOC      
gold_winter   CAN   305
              URS   250
              USA   166
silver_winter USA   308
              CAN   199
              NOR   165
bronze_winter FIN   215
              SWE   177
              USA   161
gold_summer   USA  2472
              URS   832
              GBR   636
silver_summer USA  1333
              GBR   729
              URS   635
bronze_summer USA  1197
              GER   649
              GBR   620

**Answer**

Table above is the answer. 

`CAN` - Canada 

`URS` - Soviet Union

`USA` - USA

`NOR` - Norway

`FIN` - Finland

`SWE` - Sweden

`GBR` - Great Britain

`GER` - Germany 

##12. Standart Height.##

In [79]:
corrected_data = corrected_data.assign(Height_z_scores=stats.zscore(corrected_data['Height'], nan_policy='omit'))
corrected_data

ID                          Name Sex   Age  Height  Weight  \
0        79588         Rodolfo Mijares Garza   M  22.0   180.0    75.0   
1        79589              Zlatica Mijatovi   F  26.0     NaN     NaN   
2        79590                  Pieter Mijer   M  47.0     NaN     NaN   
3        79591                     Vasa Miji   M  27.0   186.0    80.0   
4        79591                     Vasa Miji   M  31.0   186.0    80.0   
...        ...                           ...  ..   ...     ...     ...   
271110  102340           Sigrid Romero Duque   F  19.0   164.0    57.0   
271111  102340           Sigrid Romero Duque   F  19.0   164.0    57.0   
271112  102341        Susana Romero Steensma   F  17.0   173.0    64.0   
271113  102342  Yasmani Jess Romero Rodrguez   M  23.0     NaN     NaN   
271114  102343          Yoel Romero Palacios   M  23.0   175.0    84.0   

                         Team  NOC        Games    Year  Season       City  \
0                      Mexico  MEX  1960 Summer  1960.0  Summer       Roma   
1                  Yugoslavia  YUG  1948 Summer  1948.0  Summer     London   
2                 Netherlands  NED  1928 Summer  1928.0  Summer  Amsterdam   
3       Serbia and Montenegro  SCG  2000 Summer  2000.0  Summer     Sydney   
4       Serbia and Montenegro  SCG  2004 Summer  2004.0  Summer     Athina   
...                       ...  ...          ...     ...     ...        ...   
271110               Colombia  COL  2008 Summer  2008.0  Summer    Beijing   
271111               Colombia  COL  2008 Summer  2008.0  Summer    Beijing   
271112                  Spain  ESP  2008 Summer  2008.0  Summer    Beijing   
271113                   Cuba  CUB  2012 Summer  2012.0  Summer     London   
271114                   Cuba  CUB  2000 Summer  2000.0  Summer     Sydney   

                Sport                                         Event   Medal  \
0           Athletics                     Athletics Men's Decathlon     NaN   
1          Gymnastics            Gymnastics Women's Team All-Around     NaN   
2             Fencing                Fencing Men's epee, Individual     NaN   
3          Volleyball                   Volleyball Men's Volleyball    Gold   
4          Volleyball                   Volleyball Men's Volleyball     NaN   
...               ...                                           ...     ...   
271110        Archery                    Archery Women's Individual     NaN   
271111        Archery                          Archery Women's Team     NaN   
271112        Sailing             Sailing Women's One Person Dinghy     NaN   
271113  Weightlifting              Weightlifting Men's Bantamweight     NaN   
271114      Wrestling  Wrestling Men's Light-Heavyweight, Freestyle  Silver   

        Height_z_scores  
0              0.443126  
1                   NaN  
2                   NaN  
3              1.013551  
4              1.013551  
...                 ...  
271110        -1.078005  
271111        -1.078005  
271112        -0.222369  
271113              NaN  
271114        -0.032227  

[271113 rows x 16 columns]

##13. Scaled height.##

Firstly, we should find difference between maximal and minimal hight.

In [77]:
max_height = max(corrected_data['Height'])
min_height = min(corrected_data['Height'])
diff = max_height - min_height


In [ ]:
corrected_data = corrected_data.assign(Height_min_max_scaled=lambda x: (x['Height'] - min_height) / diff if x['Height'] is not np.NaN else np.NaN)
corrected_data

##14. Height, weight and age of Olimpic games partisipants.##

In [105]:
winter_games = corrected_data[corrected_data['Season']=='Winter']
height_women = winter_games[winter_games['Sex']=='F'].Height.dropna()
weight_women = winter_games[winter_games['Sex']=='F'].Weight.dropna()
age_women = winter_games[winter_games['Sex']=='F'].Age.dropna()
height_men = winter_games[winter_games['Sex']=='M'].Height.dropna()
weight_men = winter_games[winter_games['Sex']=='M'].Weight.dropna()
age_men = winter_games[winter_games['Sex']=='M'].Age.dropna()

In [121]:
def find_ci(sample):
    std = sample.std()
    mean_sample = np.mean(sample)
    se = std / np.sqrt(len(sample))
    left = mean_sample - 1.96 * se
    right = mean_sample + 1.96 * se
    return f'mean = {round(mean_sample, 2)}; conf.int. = ({round(left, 2)}-{round(right, 2)})'


In [122]:
height_women_interval = find_ci(height_women)
weight_women_interval = find_ci(weight_women)
age_women_interval = find_ci(age_women)
height_men_interval = find_ci(height_men)
weight_men_interval = find_ci(weight_men)
age_men_interval = find_ci(age_men)

In [123]:
data_for_publ = {'Mean height' : [height_men_interval, height_women_interval, round(height_men.mean() - height_women.mean(),2)],
                 'Mean weight' : [weight_men_interval, weight_women_interval, round(weight_men.mean() - weight_women.mean(),2)],
                 'Mean age' : [age_men_interval, age_women_interval, round(age_men.mean() - age_women.mean(),2)]}

table_for_publ = pd.DataFrame(data_for_publ, index=['Men', 'Women', 'Difference'])

table_for_publ

Mean height  \
Men         mean = 178.67; conf.int. = (178.59-178.75)   
Women       mean = 166.53; conf.int. = (166.43-166.63)   
Difference                                       12.14   

                                        Mean weight  \
Men         mean = 76.36; conf.int. = (76.23-76.48)   
Women       mean = 59.76; conf.int. = (59.64-59.88)   
Difference                                     16.6   

                                           Mean age  
Men          mean = 25.5; conf.int. = (25.45-25.56)  
Women       mean = 24.01; conf.int. = (23.94-24.09)  
Difference                                     1.49

Everywhere (for hight, weight and age) we have no intersections of confidence interval for men and women. Mean values and difference of this values for men and women represented in table above. It's possible to copy this table to paper or export excel:
```
table_for_publ.to_excel(f'{path}', index=True)
```

##15. Team and Medal relation.##

Amount of medals depends on team. We saw this in #11.

We can try to look at this dependence in terms of correlation coeffitient, but it is really rough estimation, because 'Medal' is categorial and can have only three values. That's why coefficient is not so big.


In [133]:
corrected_data['NOC'].astype('category').cat.codes.corr(corrected_data['Medal'].astype('category').cat.codes, method='spearman')

0.0649065000179632

Maybe if we count the probabilyty for athlet from certain country to have certain type of medal we can say more about Team and Medal relation.